src/persist-client/src/impl/state.rs

```diff
@@ -309,14 +313,46 @@ where
         self.collections.trace.upper().clone()
     }
 
-    pub fn seqno_since(&self) -> SeqNo {
+    pub fn batch_count(&self) -> usize {
+        self.collections.trace.num_hollow_batches()
+    }
+
+    fn seqno_since(&self) -> SeqNo {
         let mut seqno_since = self.seqno;
         for cap in self.collections.readers.values() {
             seqno_since = std::cmp::min(seqno_since, cap.seqno);
         }
         seqno_since
     }
 
+    pub fn maybe_gc(&mut self, is_write: bool) -> Option<(SeqNo, SeqNo)> {
```

Comment:
nit: could document that this will not just determine whether we should do a gc but also records this in state immediately.